src/test/java/seedu/address/logic/commands/patient/DeletePatientCommandTest.java

```diff
@@ -27,69 +31,109 @@
 
-        Patient personToDelete = model.getFilteredPatientList().get(INDEX_FIRST_IN_LIST.getZeroBased());
+        Patient personToDelete = emptyApptScheduleModel.getFilteredPatientList().get(
+                INDEX_FIRST_IN_LIST.getZeroBased());
         DeletePatientCommand deletePatientCommand = new DeletePatientCommand(INDEX_FIRST_IN_LIST, false);
 
-        String expectedMessage = String.format(Messages.MESSAGE_DELETE_PATIENT_SUCCESS, personToDelete);
+        String expectedMessage = String.format(MESSAGE_DELETE_PATIENT_SUCCESS, personToDelete);
 
         Model expectedModel = new ModelManager(
-                new AddressBook<>(model.getPatientRecords()),
-                new AddressBook<>(model.getDoctorRecords()),
-                new AppointmentSchedule(model.getAppointmentSchedule()),
-                new UserPrefs(model.getUserPrefs())
+                new AddressBook<>(emptyApptScheduleModel.getPatientRecords()),
+                new AddressBook<>(emptyApptScheduleModel.getDoctorRecords()),
+                new AppointmentSchedule(emptyApptScheduleModel.getAppointmentSchedule()),
+                new UserPrefs(emptyApptScheduleModel.getUserPrefs())
         );
 
         expectedModel.deletePatient(personToDelete);
         showNoPerson(expectedModel);
 
-        assertCommandSuccess(deletePatientCommand, model, expectedMessage, expectedModel);
+        assertCommandSuccess(deletePatientCommand, emptyApptScheduleModel, expectedMessage, expectedModel);
     }
 
     @Test
-    public void execute_invalidIndexFilteredList_throwsCommandException() {
-        showPatientAtIndex(model, INDEX_FIRST_IN_LIST);
+    public void execute_nonForceInvalidIndexFilteredPatientRecordsNoExistingAppointment_throwsCommandException() {
+        showPatientAtIndex(emptyApptScheduleModel, INDEX_FIRST_IN_LIST);
 
         Index outOfBoundIndex = INDEX_SECOND_IN_LIST;
         // ensures that outOfBoundIndex is still in bounds of address book list
-        assertTrue(outOfBoundIndex.getZeroBased() < model.getPatientRecords().getPersonList().size());
+        assertTrue(outOfBoundIndex.getZeroBased() <
+                emptyApptScheduleModel.getPatientRecords().getPersonList().size());
 
         DeletePatientCommand deletePatientCommand = new DeletePatientCommand(outOfBoundIndex, false);
 
-        assertCommandFailure(deletePatientCommand, model, Messages.MESSAGE_INVALID_PATIENT_DISPLAYED_INDEX);
+        assertCommandFailure(deletePatientCommand, emptyApptScheduleModel, MESSAGE_INVALID_PATIENT_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_forceValidIndexUnfilteredPatientRecordsExistingAppointment_success() {
+        Model nonEmptyApptScheduleModel = new ModelManager(getTypicalPatientRecords(), getTypicalDoctorRecords(),
+                getTypicalAppointmentSchedule(), new UserPrefs());
+
+        Patient patientToDelete = nonEmptyApptScheduleModel.getFilteredPatientList().get(
+                INDEX_SECOND_IN_LIST.getZeroBased());
```

Comment:
wouldn't this be the same as `INDEX_FIRST_IN_LIST.getOneBased()`?

Reply:
I removed the appointments for Alice a long time ago so I used second in list instead. I'm going to add back Alice's appointment, but this testcase would work still.